model/modeldecoder/v2/decoder.go

```diff
@@ -698,6 +729,20 @@ func mapToServiceModel(from contextService, out *model.Service) {
 	if from.Version.IsSet() {
 		out.Version = from.Version.Val
 	}
+	outOrigin := out.Origin
+	if outOrigin == nil {
+		outOrigin = &model.ServiceOrigin{}
+	}
```

Comment:
we should probably only set this `if from.Origin.IsSet()`
same for cloud and faas